### PR TITLE
Add option to store raw evidence binary content

### LINKED ISSUE #87: Add support for binary evidence
## Overview

We should add support for gathering binary evidence.  A need arose for fetching excel spreadsheets.

## Requirements

- ~~Add BinaryEvidence~~
   - ~~To be used with fetching evidence only~~
   - ~~Add appropriate decorator~~
   - ~~Add appropriate context manager~~
- Add a binary content flag to RawEvidence
- Update locker to write binary content

## Approach

TBD

## Security and Privacy

N/A

## Test Plan

TBD

### PR DESCRIPTION
**Pull request by alfinkel**
- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)

## What

- Add option to store raw evidence binary content

## Why

To allow for the storage of binary content in the evidence locker such as excel files.

## How

- Added the `binary_content` attribute to RawEvidence
- If set then have the locker write the evidence file as a binary data stream.
- Added docs
- Also cleaned up some doc wording

## Test

- Current functionality is not compromised
- Writing of a JPEG image to the locker as part of a fetch was successful.

## Context

Closes #87

### CODE REVIEW

**Review by cletomartin — APPROVED** [review]
Nice! +1

File: compliance/locker.py [hunk 1]
```diff
@@ -222,7 +222,8 @@ def add_evidence(self, evidence, checks=None, evidence_used=None):
                     f.write(evidence.get_partition(key))
         else:
             full_path = os.path.join(self.local_path, evidence.path)
```
Comment (drsm79) [inline compliance/locker.py:224]: Using pathlib here might be nice. You can write o the Path, too.

**Comment by drsm79** [comment]
I’m fairly sure there was an image retrieved by the old demo fetcher without issue. Was this triggered by a fetcher that was stricter about binary content? Or maybe a python 3 thing?

**Comment by alfinkel** [comment]
Now that you mention it I do now remember that.  Although it was a while ago now.  This was triggered by https://github.com/ComplianceAsCode/auditree-framework/issues/87#issuecomment-732635163.

File: compliance/locker.py (hunk 1 above)
Comment (alfinkel) [reply compliance/locker.py:224]: https://github.com/ComplianceAsCode/auditree-framework/issues/89

**Comment by drsm79** [comment]
Cool 😎

File: compliance/locker.py (hunk 1 above)
Comment (drsm79) [reply compliance/locker.py:224]: Nice one, thanks

**Pull request merged**